src/notebooks/student_logins.ipynb

```diff
@@ -0,0 +1,277 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import os\n",
+    "import lms_utilities as lms\n",
+    "import matplotlib as plt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# set the directory path for the output files from schoology extractor\n",
+    "csv_path = os.path.join(\"..\", \"schoology-extractor\", \"data\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# pull user data into a dataframe\n",
+    "users_df = lms.get_all_users(csv_path)\n",
+    "\n",
+    "# pull login data into a dataframe\n",
+    "logins_df = lms.get_all_system_activities(csv_path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# merge user and login data for schoology\n",
+    "user_logins_df = pd.DataFrame()\n",
+    "user_logins_df = pd.merge(users_df, logins_df, how='inner', left_on=('SourceSystemIdentifier', 'SourceSystem'), right_on=('LMSUserSourceSystemIdentifier', 'SourceSystem'))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# add user and activity columns to login frequency dataframe\n",
+    "login_df = pd.DataFrame()\n",
+    "login_df = user_logins_df.loc[:,['Name', 'EmailAddress', 'SISUserIdentifier','ActivityType']].query(\"ActivityType not in ['sign-out']\")\n",
+    "\n",
+    "# drop ActivityType column\n",
+    "del login_df['ActivityType']\n",
```

Comment:
This works just fine. I _think_ - but not 100% sure - that it would be more efficient to use the `drop` function though:

```suggestion
    "login_df.drop(columns=['ActivityType'], inplace=True)\n",
```

Reply:
Interestingly, my assumption is not correct. For the code below, I get the following timings:

* `del` -> 737 microseconds
* `drop` -> 1.01 milliseconds
* `drop(inplace=True)` -> 1 milleseconds

No significant difference on the two `drops` but over 20% faster to use `del`. At that speed it hardly matters in this case :-D. So `del` is great. I'll probably stick with `drop` as is it idiomatically appropriate for a DataFrame.

```python
import pandas as pd
a=list()
b=list()
for i in range(100000):
    a.append(i)
    b.append(-i)

c = zip(a,b)
df = pd.DataFrame(c, columns=["a", "b"])

def _del():
    df1 = df.copy()
    del df1["b"]

def _drop():
    df2 = df.copy()
    df2 = df2.drop(columns=["b"])

def _drop_inplace():
    df3 = df.copy()
    df3.drop(columns=["b"], inplace=True)

%timeit _del()
%timeit _drop()
%timeit _drop_inplace()
```